Selecting headers from right to left causes an error

If I select the header range starting at the right most column and working left, as so:

![image](https://github.com/user-attachments/assets/aa864e2a-aa40-4c7e-8b08-abb6117c84f2)

I get an assertion error that the end of the range is before the start (which is true, but fine if normalised...)

```
Template render error: (/Users/simon/data-import/prototypes/basic/app/views/mapping.html) [Line 18, Column 34]
  AssertionError: The expression evaluated to a falsy value:

  assert(range.end.column >= range.start.column)

    at Object._prettifyError (/Users/simon/data-import/prototypes/basic/node_modules/nunjucks/src/lib.js:32:11)
    at /Users/simon/data-import/prototypes/basic/node_modules/nunjucks/src/environment.js:464:19
    at eval (eval at _compile (/Users/simon/data-import/prototypes/basic/node_modules/nunjucks/src/environment.js:527:18), <anonymous>:115:12)
    at eval (eval at _compile (/Users/simon/data-import/prototypes/basic/node_modules/nunjucks/src/environment.js:527:18), <anonymous>:260:12)
    at b_content (eval at _compile (/Users/simon/data-import/prototypes/basic/node_modules/nunjucks/src/environment.js:527:18), <anonymous>:96:3)
    at eval (eval at _compile (/Users/simon/data-import/prototypes/basic/node_modules/nunjucks/src/environment.js:527:18), <anonymous>:259:28)
    at b_beforeContent (eval at _compile (/Users/simon/data-import/prototypes/basic/node_modules/nunjucks/src/environment.js:527:18), <anonymous>:72:1)
    at b_main (eval at _compile (/Users/simon/data-import/prototypes/basic/node_modules/nunjucks/src/environment.js:527:18), <anonymous>:242:34)
    at eval (eval at _compile (/Users/simon/data-import/prototypes/basic/node_modules/nunjucks/src/environment.js:527:18), <anonymous>:114:83)
    at b_header (eval at _compile (/Users/simon/data-import/prototypes/basic/node_modules/nunjucks/src/environment.js:527:18), <anonymous>:618:1)
```

--- a/lib/importer/index.js
+++ b/lib/importer/index.js
@@ -236,10 +236,18 @@ exports.Initialise = (config, router, prototypeKit) => {
     // We should probably find a better way of handling no selection, and we still
     // need to decide how users can decide between auto-choose header and user-chosen
     // headers.
-    let tlRow = request.body['importer:selection:TLRow'] ?? 0;
-    let tlCol = request.body['importer:selection:TLCol'] ?? 0;
-    let brRow = request.body['importer:selection:BRRow'] ?? 0;
-    let brCol = request.body['importer:selection:BRCol'] ?? maxCol;
+    let tlRow = getIntOrDefault(request.body['importer:selection:TLRow']);
+    let tlCol = getIntOrDefault(request.body['importer:selection:TLCol']);
+    let brRow = getIntOrDefault(request.body['importer:selection:BRRow']);
+    let brCol = getIntOrDefault(request.body['importer:selection:BRCol'], maxCol);
+
+    // Normalise the selection so that start row <= end row and start col <= end col
+    if (tlRow > brRow) {
+      [tlRow, brRow] = [brRow, tlRow];
+    }
+    if (tlCol > brCol) {
+      [tlCol, brCol] = [brCol, tlCol];
+    }
 
     session.headerRange = {
       sheet: session.sheet,
@@ -251,8 +259,6 @@ exports.Initialise = (config, router, prototypeKit) => {
     request.session.data[IMPORTER_SESSION_KEY] = session;
     redirectOnwards(request, response);
   });
-
-
 
 
   //--------------------------------------------------------------------
@@ -273,6 +279,7 @@ exports.Initialise = (config, router, prototypeKit) => {
     request.session.data[IMPORTER_SESSION_KEY] = session;
     redirectOnwards(request, response);
   });
+
 
   //--------------------------------------------------------------------
   // Review the processing for the current session before continuing.
@@ -320,3 +327,14 @@ exports.CreateSession = session_lib.CreateSession;
 exports.GetSession = session_lib.GetSession;
 exports.UpdateSession = session_lib.UpdateSession;
 exports.ListSessions = session_lib.ListSessions;
+
+//--------------------------------------------------------------------
+// Parse the given key into an int if it is truthy, or returns the
+// default value (whose default is 0)
+//--------------------------------------------------------------------
+const getIntOrDefault = (key, dflt=0) => {
+  if (key) {
+    return parseInt(key)
+  }
+  return dflt;
+};

